### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by kunyuan**
1. Add plasma frequency and Thomas-Fermi screening length to parameter
2. Add three approximations for the 3D free-electron polarization
3. Add some tests.

### CODE REVIEW

File: src/parameter.jl [hunk 1]
```diff
@@ -44,9 +44,11 @@ using ..Parameters
     e0s::Float64 = e0
     e0a::Float64 = espin
     NF::Float64 = (dim == 3) ? spin * me * kF / 2 / π^2 : spin * me / 2 / π
+    ωp::Float64 = (dim == 3) ? sqrt(4π * e0^2 * n / me) : 0.0 # plasma frequency
+    qTF::Float64 = (dim == 3) ? sqrt(4π * e0^2 * NF) : 0.0 # inverse thomas-fermi screening length
 end
 
-derived_para_names = (:beta, :Θ, :T, :n, :Rs, :a0, :rs, :kF, :espin, :e0s, :e0a, :NF)
+derived_para_names = (:beta, :Θ, :T, :n, :Rs, :a0, :rs, :kF, :espin, :e0s, :e0a, :NF, :ωp)
```
Comment (iintSjds) [inline src/parameter.jl]: should qTF also be added here?

**Comment by iintSjds** [comment]
Got error from example/G0W0_3D.ipynb with julia 1.7.0 kernel.
Detail:
ReadOnlyMemoryError()

Stacktrace:
  [1] getrf!(A::Matrix{ComplexF64})
    @ LinearAlgebra.LAPACK ~/opt/julia/julia-1.7.0/share/julia/stdlib/v1.7/LinearAlgebra/src/lapack.jl:575
  [2] #lu!#146
    @ ~/opt/julia/julia-1.7.0/share/julia/stdlib/v1.7/LinearAlgebra/src/lu.jl:81 [inlined]
  [3] #lu#153
    @ ~/opt/julia/julia-1.7.0/share/julia/stdlib/v1.7/LinearAlgebra/src/lu.jl:279 [inlined]
  [4] lu (repeats 2 times)
    @ ~/opt/julia/julia-1.7.0/share/julia/stdlib/v1.7/LinearAlgebra/src/lu.jl:278 [inlined]
  [5] \(A::Matrix{ComplexF64}, B::Matrix{ComplexF64})
    @ LinearAlgebra ~/opt/julia/julia-1.7.0/share/julia/stdlib/v1.7/LinearAlgebra/src/generic.jl:1142
  [6] _weightedLeastSqureFit(dlrGrid::Lehmann.DLRGrid, Gτ::Matrix{ComplexF64}, error::Nothing, kernel::Matrix{ComplexF64}, sumrule::Nothing)
    @ Lehmann ~/.julia/packages/Lehmann/uKJiS/src/operation.jl:159
  [7] matfreq2dlr(dlrGrid::Lehmann.DLRGrid, green::Array{ComplexF64, 4}, nGrid::Vector{Int64}; error::Nothing, axis::Int64, sumrule::Nothing, verbose::Bool)
    @ Lehmann ~/.julia/packages/Lehmann/uKJiS/src/operation.jl:328
  [8] matfreq2tau(dlrGrid::Lehmann.DLRGrid, green::Array{ComplexF64, 4}, τNewGrid::Vector{Float64}, nGrid::Vector{Int64}; error::Nothing, axis::Int64, sumrule::Nothing, verbose::Bool)
    @ Lehmann ~/.julia/packages/Lehmann/uKJiS/src/operation.jl:410
  [9] toTau(green::Green2DLR{ComplexF64, ImFreq, CompositeGrids.SimpleG.Arbitrary{Int64}, CompositeGrids.CompositeG.Composite{Float64, CompositeGrids.SimpleG.Arbitrary{Float64}, CompositeGrids.CompositeG.Composite{Float64, CompositeGrids.SimpleG.Log{Float64}, CompositeGrids.SimpleG.BaryCheb{Float64}}}}, targetGrid::Vector{Float64})
    @ GreenFunc ~/.julia/packages/GreenFunc/lRJz0/src/green/Green.jl:225
 [10] toTau
    @ ~/.julia/packages/GreenFunc/lRJz0/src/green/Green.jl:209 [inlined]
 [11] calcΣ_3d(G::Green2DLR{ComplexF64, ImFreq, CompositeGrids.SimpleG.Arbitrary{Int64}, CompositeGrids.CompositeG.Composite{Float64, CompositeGrids.SimpleG.Arbitrary{Float64}, CompositeGrids.CompositeG.Composite{Float64, CompositeGrids.SimpleG.Log{Float64}, CompositeGrids.SimpleG.BaryCheb{Float64}}}}, W::ElectronGas.LegendreInteraction.DCKernel)
    @ ElectronGas.SelfEnergy /media/iintsjds/File/Research/JuliaLibs/ElectronGas.jl/src/selfenergy.jl:182
 [12] G0W0(param::ElectronGas.Parameter.Para, Euv::Float64, rtol::Float64, Nk::Int64, maxK::Float64, minK::Float64, order::Int64, int_type::Symbol; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ ElectronGas.SelfEnergy /media/iintsjds/File/Research/JuliaLibs/ElectronGas.jl/src/selfenergy.jl:233
 [13] G0W0
    @ /media/iintsjds/File/Research/JuliaLibs/ElectronGas.jl/src/selfenergy.jl:218 [inlined]
 [14] #G0W0#2
    @ /media/iintsjds/File/Research/JuliaLibs/ElectronGas.jl/src/selfenergy.jl:242 [inlined]
 [15] top-level scope
    @ In[5]:2
 [16] eval
    @ ./boot.jl:373 [inlined]
 [17] include_string(mapexpr::typeof(REPL.softscope), mod::Module, code::String, filename::String)
    @ Base ./loading.jl:1196

**Comment by iintSjds** [comment]
Previous error resolved. Got a new error from G0W0_3D.ipynb under "1.2 Momentum-dependent Self-energy (instant + dynamic)":
MethodError: no method matching zfactor(::Green2DLR{ComplexF64, ImFreq, CompositeGrids.SimpleG.Arbitrary{Int64}, CompositeGrids.CompositeG.Composite{Float64, CompositeGrids.SimpleG.Arbitrary{Float64}, CompositeGrids.CompositeG.Composite{Float64, CompositeGrids.SimpleG.Log{Float64}, CompositeGrids.SimpleG.BaryCheb{Float64}}}})

Guess it's from 
z=SelfEnergy.zfactor(sigma_wn)
it should now be something like
z=SelfEnergy.zfactor(param, sigma_wn)

**Comment by kunyuan** [comment]
> Previous error resolved. Got a new error from G0W0_3D.ipynb under "1.2 Momentum-dependent Self-energy (instant + dynamic)": MethodError: no method matching zfactor(::Green2DLR{ComplexF64, ImFreq, CompositeGrids.SimpleG.Arbitrary{Int64}, CompositeGrids.CompositeG.Composite{Float64, CompositeGrids.SimpleG.Arbitrary{Float64}, CompositeGrids.CompositeG.Composite{Float64, CompositeGrids.SimpleG.Log{Float64}, CompositeGrids.SimpleG.BaryCheb{Float64}}}})
> 
> Guess it's from z=SelfEnergy.zfactor(sigma_wn) it should now be something like z=SelfEnergy.zfactor(param, sigma_wn)

The problem should be fixed

**Review by iintSjds — APPROVED** [review]

**Pull request merged**